### ISSUE TITLE
Buck requires python 2.7

### ISSUE THREAD

**Issue by hey99xx** [facebook/buck#1381]
So although it's advertised as buck supporting python 2.6 and 2.7, I've just found out it's not the case.

In https://github.com/facebook/buck/blob/master/programs/buck_repo.py#L10 you're using 

    from subprocess import check_output

which is not available in 2.6. So I am ok using 2.7 for python. But the issue is that I cannot force buck to pick up Python 2.7 from my system. Buck's bin file is written in a way that makes it impossible https://github.com/facebook/buck/blob/master/bin/buck#L21

    PYTHON=$(command -v python2 python | head -1)

On my system python2 is of version 2.6 coming from /usr/bin/python2, and python is of version 2.7 coming from a custom installed location. But you're using `command` which skips over my PATH and chooses first python2, which is 2.6 and now my builds are broken with

```
~/Desktop/downloads/buck $ ./bin/buck --help
Traceback (most recent call last):
  File "/local/home/user/Desktop/downloads/buck/bin/../programs/buck.py", line 48, in <module>
    propagate_failure(main(sys.argv))
  File "/local/home/user/Desktop/downloads/buck/bin/../programs/buck.py", line 36, in main
    with get_repo(project) as buck_repo:
  File "/local/home/user/Desktop/downloads/buck/bin/../programs/buck.py", line 25, in get_repo
    from buck_repo import BuckRepo
  File "/local/home/user/Desktop/downloads/buck/programs/buck_repo.py", line 10, in <module>
    from subprocess import check_output
ImportError: cannot import name check_output
```

**Comment by Coneko** [comment]
It should be using `check_output` from here: https://github.com/facebook/buck/blob/master/programs/subprocutils.py#L19.

**Comment by Coneko** [comment]
cc @yiding

**Comment by yiding** [comment]
I think we should either test buck with python 2.6 or drop support for it. Meanwhile, I'll fix this.

**Comment by yiding** [comment]
> But you're using command which skips over my PATH and chooses first python2, which is 2.6

Can you elaborate here? `command` respects your PATH unless `-p` is given. From the Bash man page:

```
command [-pVv] command [arg ...]
       Run command with args  suppressing  the  normal  shell  function
       lookup.  Only builtin commands or commands found in the PATH are
       executed.  If the -p option is given, the search for command  is
       performed  using  a default value for PATH that is guaranteed to
       find all of the standard utilities.  If  either  the  -V  or  -v
       option is supplied, a description of command is printed.
```

**Comment by hey99xx** [comment]
@yiding sorry if I misunderstood `command`'s explanation from a coworker. It may be using `PATH` which is fine. But the thing is buck would choose python 2.6 regardless. On my system:

```
$ command -v python2 python
/usr/bin/python2
/mycustomlocation/python

$ /usr/bin/python2 --version
Python 2.6.9

$ /mycustomlocation/python --version
Python 2.7.8
```

which means `PYTHON=$(command -v python2 python | head -1)` will always be Python 2.6, unless I hack around my system and create a `python2` symlink and point it to 2.7. But Python 2.6 came out of the box in my Linux, so I'm hesitant to modify my path to use Buck. As a temporary workaround, I've rewritten that line as  `PYTHON=$(command -v python | head -1)` so I can keep building my dependent projects.

Thanks for the quick look at this issue.

**Comment by yiding** [comment]
Fixed in b2af4022fe7168a4a31571a771ebda426af702eb.
